aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+
+The introduction of Sign in with Aptos is transformative for the ecosystem, as it enables developers to create robust backends tied to secure, blockchain-based authentication. It opens doors for operations that don’t need to be on the blockchain, such as storing profile information, rate limiting, and more.
+
+## Alternative Solutions
+
+> Explain why you submitted this proposal specifically over alternative solutions. Why is this the best possible outcome?
+
+The main alternative is to continue using the `connect` + `signMessage` flow. This flow is not as secure as SIWA, exposing users to the risk of phishing attacks and a poor double prompt user experience.
+
+![Figure 2: Connect and Sign Message Flow](./static/9.png)
+
+## Specification and Implementation Details
+
+The new standard introduces new interfaces that conform to CAIP-122. Wallets will be responsible for the construction of the messages. Applications will be responsible for providing signing requests. Three interfaces will be introduced: `AptosSignInInput`, `AptosSignInOutput`, and `AptosSignInMessage`.
+
+### AptosSignInInput
+
+The `AptosSignInInput` interface will be part of the signing request to construct the SIWA message. These fields will be sent from the application to the wallet.
+
+| Name           | Type            | Mandatory | Description                                                                                                                           |
+| -------------- | --------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| domain         | string          |           | `dnsauthority` that is requesting the signing.                                                                                        |
+| address        | string          |           | Blockchain address performing the signing.                                                                                            |
+| uri            | string          |           | URI referring to the resource that is the subject of the signing i.e. the subject of the claim.                                       |
+| version        | string          |           | Current version of the message.                                                                                                       |
+| statement      | string          |           | Human-readable ASCII assertion that the user will sign. It MUST NOT contain `\n`.                                                     |
+| nonce          | string          |           | Randomized token to prevent signature replay attacks.                                                                                 |
+| chainId        | string          |           | Identifier for the network locating the address listed separately above.                                                              |
```

Comment:
_"Identifier for the network locating the address listed separately above"_
--> 
"Identifier for the network where the `address` above lives (e.g., for Aptos `mainnet`, the chain ID would be 1)"